### DISCUSSION TITLE
Finding the best parameters

### DISCUSSION THREAD

**Discussion by mandar-avhad** [explosion/spaCy#11809] [Help: Model Advice] [answered]
Hello,
Is there any way to generate the best parameters while training Spacy NER model?
Something like grid search?
I want to automate the training process by training the model with the best parameters in each particular case.
Please share some references or examples if any.
Thank you

**Comment by thomashacker** [comment, marked as answer]
Hello, 
you can use either use spaCy's `wandb` ([docs](https://docs.wandb.ai/guides/integrations/spacy)) or `ClearML` ([docs](https://github.com/explosion/spacy-loggers#clearmllogger)) integration to do parameter optimization in spaCy. 
For `wandb`, we provide a [spaCy project](https://github.com/explosion/projects/tree/v3/integrations/wandb) showing how to setup parameter search.

**Reply by mandar-avhad** [reply]
Thanks.
How can I integrate it with the Spacy training code/cli command?
